### PR TITLE
Web:Discover Refactor resource selector

### LINKED ISSUE #22128: [Discover] Day 2 "Enroll new resources" screen
As part of Day 2, the "Enroll new resources" section needs to be re-implemented (https://www.figma.com/file/uLevdNsEnIvvLDSZ9sQqXM/Teleport-Discover-Access?node-id=182%3A12030&t=MEPONWCs2gublJIp-0)

We should tag each resource with as much information as possible, to allow for the user to search for a variety of things ("database" or "db" should show all database resources, "aws" should show AWS resources, etc).

### PR DESCRIPTION
**Pull request by kimlisa**
resolves https://github.com/gravitational/teleport/issues/22128

Please don't hate me 😓, one thing led to another and I ended up with a huge PR (I feel like most touched files are from deletion, clean up, rename, just micro changes)

I recommend reviewing by commit, I spent some time to make each commit contained and added detailed commit messages if applicable.

#### Description
- Refactors the `Select a resource` screen. Before each resource type had its own section, now all resources are combined into one view (which was why we had huge deletion section)
   - I listed every database that i found on the docs (as requested), and the resources that are not guided, we will just link user out to the docs
   - allows users to filter by search keywords, during meeting discussion we decided to not add a sort (filter?) dropdown at this time
- Allows user to go back to the select resource screen on the first view

#### Screenshot

https://user-images.githubusercontent.com/43280172/225269878-2e0dcba2-035b-4fe5-b116-4d94810e9072.mov

<br/>
when user does not have any access

image

permission detail

image

### CODE REVIEW

**Comment by kimlisa** [comment]
friendly ping @ryanclark @ibeckermayer

**Review by ibeckermayer — APPROVED** [review]
Looking good especially for such a large change.

**Review by ryanclark — APPROVED** [review]
LGTM, just a minor comment

**Comment by ryanclark** [comment]
Just ran this locally and these icons aren't aligned - 

image

File: web/packages/teleport/src/Discover/SelectResource/PermissionsErrorMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Text, Box } from 'design';
+
+import { ResourceKind } from '../Shared';
+
+export function PermissionsErrorMessage({
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleport/src/Discover/SelectResource/PermissionsErrorMessage.tsx:22]: hm, the component itself is not a tool tip, it only contains perm error message

File: web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx [hunk 2]
```diff
@@ -15,206 +15,431 @@
...
+      sorted = [
+        ...resources.filter(r => r.kind === ResourceKind.Kubernetes),
+        ...resources.filter(r => r.kind !== ResourceKind.Kubernetes),
+      ];
+      break;
+  }
+  return sorted;
+}
+
+function checkAndSetHasNoAccessFieldToResources(acl: Acl) {
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx]: renamed to `makeResourcesWithHasAccessField` and renamed to `hasAccess`

File: web/packages/teleport/src/Discover/SelectResource/types.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export enum DatabaseLocation {
+  AWS,
+  SelfHosted,
+  GCP,
+  Azure,
+  Snowflake,
+  Mongo,
+
+  TODO,
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleport/src/Discover/SelectResource/types.ts:27]: just temp filler for locations and engines that don't use these enums yet but is required to define the field

**Pull request merged**